Gateway/Request/CustomerIpDataBuilder.php

```diff
@@ -52,7 +53,12 @@ public function build(array $buildSubject)
     {
         /** @var \Magento\Payment\Gateway\Data\PaymentDataObject $paymentDataObject */
         $paymentDataObject = \Magento\Payment\Gateway\Helper\SubjectReader::readPayment($buildSubject);
-        $shopperIp = $paymentDataObject->getPayment()->getOrder()->getXForwardedFor();
+        $order = $paymentDataObject->getPayment()->getOrder();
+        $shopperIp = $order->getXForwardedFor();
+
+        if (empty($shopperIp)) {
+            $shopperIp = $order->getRemoteIp();
```

Comment:
are we sure that this does not result in a server IP ? because if that is the case it is better to not send a value!

Reply:
Without ip address sent, afterpay results in an internal server error so looks like shopperIp is a required field.
Can we maybe send like a default value in this case, does it make sense?